portal-impl/src/com/liferay/portal/language/LanguageImpl.java

```diff
@@ -1813,8 +1818,9 @@ else if (argument instanceof Date) {
 			_groupLanguageIdLocalesMap.get(groupId);
 
 		if (groupLanguageIdLocalesMap == null) {
-			ObjectValuePair<HashMap<String, Locale>, HashMap<String, Locale>>
-				objectValuePair = _createGroupLocales(groupId);
+			ObjectValuePair<LinkedHashMap<String, Locale>,
+				LinkedHashMap<String, Locale>>
+					objectValuePair = _createGroupLocales(groupId);
```

Comment:
we may not use LinkedHashMap because LinkedHashMap  extends HashMap.

Reply:
HashMap is unordered, but Linked HashMap is ordered ant it extends HashMap, so I think it can be used to replace HashMap here to meet customer's requirement.

Reply:
At here, if we use HashMap, it will work. Because you set LinkedHashMap on https://github.com/Seiphon/liferay-portal/blob/65770123356b23913b7caea50803f2a32860f42f/portal-impl/src/com/liferay/portal/language/LanguageImpl.java#L1637-L1640